src/services/kratos/auth-phone-no-password.ts

```diff
@@ -141,6 +141,42 @@ export const AuthWithPhonePasswordlessService = (): IAuthWithPhonePasswordlessSe
     return kratosUserId
   }
 
+  const updatePhone = async ({
+    kratosUserId,
+    phone,
+  }: {
+    kratosUserId: UserId
+    phone: PhoneNumber
+  }) => {
+    let identity: Identity
+
+    try {
+      ;({ data: identity } = await kratosAdmin.getIdentity({ id: kratosUserId }))
+    } catch (err) {
+      return new UnknownKratosError(err)
+    }
+
+    if (identity.state === undefined)
+      throw new KratosError("state undefined, probably impossible state") // type issue
+
+    identity.traits = { phone }
+
+    const adminIdentity: UpdateIdentityBody = {
+      ...identity,
+      credentials: { password: { config: { password } } },
+      state: identity.state,
+    }
+
+    const { data: newIdentity } = await kratosAdmin.updateIdentity({
```

Comment:
This can probably throw? If yes, we should wrap in try-catch (may also be an issue in existing `upgradeToPhoneAndEmailSchema` method)

Reply:
Fixed